### DISCUSSION TITLE
What is the point of tensor merging?

### DISCUSSION THREAD

**Discussion by miasik** [hako-mikan/sd-webui-supermerger#146] [Q&A]
I've read about the tensor merging in https://github.com/hako-mikan/sd-webui-supermerger/blob/main/calcmode_en.md#tensor but can't get the point.
What is the goal of the tensor merging? What are its advantages and disadvantages?

**Comment by hako-mikan** [comment]
As we know, in a normal merge, the tensors of Model A and Model B are mixed according to their ratios; the advantage of tensor merging is that the original tensors of each model are preserved.
I expect that the purpose of tensor merging is to reduce the amount of information that is lost by merging, but I have not had time to verify this and have not tried it.

**Reply by miasik** [reply]
I've got the goal reading your documentation but I can't get the way. As I see, if we want keep tensors whole, we don't have to use %. We should use number/s of tensors or ranges. Something like 1,2,4-6,134-677
Correct me, if I'm wrong. If the tensors are the same as elements, we can use names and ranges too. I mean that

```
output_blocks.4.0.out_layers.0.weight:
output_blocks.4.0.skip_connection.weight:
output_blocks.4.1.transformer_blocks.0.ff.net.0.proj:
output_blocks.4.1.transformer_blocks.0.ff.net.2.bias:
```
tells the system to directly replace the elements from A to B without any merging.

**Reply by hako-mikan** [reply]
In a regular merge, if the elements are further subdivided and merged at a ratio of 0.5, the resulting tensor becomes the sum of the tensor multiplied by 0.5.
In tensor merge, a new tensor is created that contains half of the tensor from model A and half from model B.

**Reply by miasik** [reply]
Oh, got it. The illustration confused me so much :-(
What is the "tensor2"?

**Reply by TomLucidor** [reply]
@hako-mikan but then how do they select the tensors for each model when putting them together? What if they are already pretty optimized each and thus would probably lose vital information dropping some of them? Or am I missing something?
